kube/pod.go

```diff
@@ -48,7 +48,9 @@ func NewPodTemplate(role *model.InstanceGroup, settings ExportSettings, grapher
 		// This role requires a custom service account
 		spec.Add("serviceAccountName", role.Run.ServiceAccount, authModeRBAC(settings))
 	}
-
+	if settings.CreateHelmChart {
+		spec.Get("imagePullSecrets").Set(helm.Block(`if ne .Values.kube.registry.username ""`))
+	}
```

Comment:
I'm regretting that we didn't just pass `settings.CreateHelmChart` to `helm.Encoder` instead; if we did that we can probably push a bunch of the random kube logic to one place…

Reply:
Yeah, me too! Because I forgot initially to wrap the `helm.Block()` calls in the `settings.CreateHelmChart` conditional and spent some time figuring out why the tests failed.

On the other hand, it forces you to think about what to do for the plain kube files. I really wish we would get rid of them and deal with the tests in a different way. Then the whole issue would go away.